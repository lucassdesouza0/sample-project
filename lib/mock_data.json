[
    {
        "id": 1,
        "first_name": "Florinda",
        "last_name": "Le Breton De La Vieuville",
        "email": "flebretondelavieuville0@walmart.com",
        "gender": "Female",
        "address": "PO Box 28243",
        "phone": "596-668-5360",
        "car": "Bravada",
        "close_person": "303-182-0938"
    },
    {
        "id": 2,
        "first_name": "Elaine",
        "last_name": "Abdie",
        "email": "eabdie1@walmart.com",
        "gender": "Female",
        "address": "PO Box 63602",
        "phone": "955-997-2087",
        "car": "riolet",
        "close_person": "334-527-1455"
    },
    {
        "id": 3,
        "first_name": "Misti",
        "last_name": "Belvin",
        "email": "mbelvin2@msu.edu",
        "gender": "Female",
        "address": "PO Box 64590",
        "phone": "106-727-4501",
        "car": "Murano",
        "close_person": "263-843-3418"
    },
    {
        "id": 4,
        "first_name": "Pietra",
        "last_name": "Bickerdicke",
        "email": "pbickerdicke3@so-net.ne.jp",
        "gender": "Agender",
        "address": "13th Floor",
        "phone": "261-590-3092",
        "car": "Topaz",
        "close_person": "350-250-2084"
    },
    {
        "id": 5,
        "first_name": "Jere",
        "last_name": "Wohlers",
        "email": "jwohlers4@woothemes.com",
        "gender": "Male",
        "address": "Apt 1192",
        "phone": "838-339-3111",
        "car": "Continental",
        "close_person": "466-164-3112"
    },
    {
        "id": 6,
        "first_name": "Koressa",
        "last_name": "Knotte",
        "email": "kknotte5@ed.gov",
        "gender": "Female",
        "address": "Apt 880",
        "phone": "226-103-1061",
        "car": "Breeze",
        "close_person": "187-311-2098"
    },
    {
        "id": 7,
        "first_name": "Micheline",
        "last_name": "Maryan",
        "email": "mmaryan6@economist.com",
        "gender": "Genderfluid",
        "address": "Suite 49",
        "phone": "143-342-5081",
        "car": "Sundance",
        "close_person": "520-907-2467"
    },
    {
        "id": 8,
        "first_name": "Burtie",
        "last_name": "Blowing",
        "email": "bblowing7@ftc.gov",
        "gender": "Male",
        "address": "Room 138",
        "phone": "897-879-0017",
        "car": "TL",
        "close_person": "114-868-8532"
    },
    {
        "id": 9,
        "first_name": "Rodney",
        "last_name": "Rockhall",
        "email": "rrockhall8@furl.net",
        "gender": "Male",
        "address": "Apt 1098",
        "phone": "443-845-0531",
        "car": "Cabriolet",
        "close_person": "902-601-3138"
    },
    {
        "id": 10,
        "first_name": "Lonna",
        "last_name": "Sogg",
        "email": "lsogg9@amazon.co.jp",
        "gender": "Female",
        "address": "Room 768",
        "phone": "418-631-7912",
        "car": "Suburban 1500",
        "close_person": "265-123-0698"
    },
    {
        "id": 11,
        "first_name": "Cyrille",
        "last_name": "Pacey",
        "email": "cpaceya@gravatar.com",
        "gender": "Male",
        "address": "PO Box 7574",
        "phone": "948-467-4879",
        "car": "Rainier",
        "close_person": "643-389-7677"
    },
    {
        "id": 12,
        "first_name": "Brear",
        "last_name": "Riddle",
        "email": "briddleb@stanford.edu",
        "gender": "Female",
        "address": "PO Box 88697",
        "phone": "162-786-4466",
        "car": "TL",
        "close_person": "442-751-4370"
    },
    {
        "id": 13,
        "first_name": "Ingaberg",
        "last_name": "Audus",
        "email": "iaudusc@paypal.com",
        "gender": "Female",
        "address": "PO Box 72944",
        "phone": "254-146-9809",
        "car": "Lucerne",
        "close_person": "519-290-9487"
    },
    {
        "id": 14,
        "first_name": "Clayborn",
        "last_name": "Heed",
        "email": "cheedd@sfgate.com",
        "gender": "Agender",
        "address": "Room 960",
        "phone": "551-497-4463",
        "car": "RX",
        "close_person": "831-644-5763"
    },
    {
        "id": 15,
        "first_name": "Raddie",
        "last_name": "Lovejoy",
        "email": "rlovejoye@last.fm",
        "gender": "Male",
        "address": "Apt 1948",
        "phone": "580-277-0911",
        "car": "Rodeo",
        "close_person": "142-905-9695"
    },
    {
        "id": 16,
        "first_name": "Nelie",
        "last_name": "Woodyear",
        "email": "nwoodyearf@comsenz.com",
        "gender": "Female",
        "address": "Suite 2",
        "phone": "697-790-3811",
        "car": "Breeze",
        "close_person": "594-534-5455"
    },
    {
        "id": 17,
        "first_name": "Thom",
        "last_name": "MacLennan",
        "email": "tmaclennang@theatlantic.com",
        "gender": "Male",
        "address": "2nd Floor",
        "phone": "631-888-4320",
        "car": "Quest",
        "close_person": "362-703-5402"
    },
    {
        "id": 18,
        "first_name": "Buffy",
        "last_name": "Synnot",
        "email": "bsynnoth@webeden.co.uk",
        "gender": "Female",
        "address": "PO Box 69323",
        "phone": "262-603-1079",
        "car": "944",
        "close_person": "505-739-1549"
    },
    {
        "id": 19,
        "first_name": "Druci",
        "last_name": "Listone",
        "email": "dlistonei@about.me",
        "gender": "Female",
        "address": "Suite 75",
        "phone": "548-613-1495",
        "car": "FJ Cruiser",
        "close_person": "491-687-4987"
    },
    {
        "id": 20,
        "first_name": "Hal",
        "last_name": "Trebble",
        "email": "htrebblej@biglobe.ne.jp",
        "gender": "Non-binary",
        "address": "Suite 9",
        "phone": "249-647-3240",
        "car": "Safari",
        "close_person": "620-780-6393"
    },
    {
        "id": 21,
        "first_name": "Peadar",
        "last_name": "Beavon",
        "email": "pbeavonk@so-net.ne.jp",
        "gender": "Male",
        "address": "Room 351",
        "phone": "732-400-0034",
        "car": "MKX",
        "close_person": "249-878-2118"
    },
    {
        "id": 22,
        "first_name": "Nessi",
        "last_name": "Phette",
        "email": "nphettel@ifeng.com",
        "gender": "Female",
        "address": "Suite 73",
        "phone": "952-182-0395",
        "car": "XL7",
        "close_person": "193-543-0991"
    },
    {
        "id": 23,
        "first_name": "Port",
        "last_name": "Dudenie",
        "email": "pdudeniem@parallels.com",
        "gender": "Male",
        "address": "PO Box 91839",
        "phone": "104-865-4913",
        "car": "Elantra",
        "close_person": "594-635-6252"
    },
    {
        "id": 24,
        "first_name": "Maximo",
        "last_name": "Ashfold",
        "email": "mashfoldn@yandex.ru",
        "gender": "Male",
        "address": "PO Box 82774",
        "phone": "372-286-3482",
        "car": "626",
        "close_person": "376-410-0369"
    },
    {
        "id": 25,
        "first_name": "Pippy",
        "last_name": "McTavish",
        "email": "pmctavisho@vinaora.com",
        "gender": "Genderfluid",
        "address": "PO Box 48618",
        "phone": "356-675-0791",
        "car": "Ram Van B150",
        "close_person": "875-584-6719"
    },
    {
        "id": 26,
        "first_name": "Blakeley",
        "last_name": "Isaksson",
        "email": "bisakssonp@163.com",
        "gender": "Female",
        "address": "PO Box 52887",
        "phone": "143-300-4225",
        "car": "Dakota",
        "close_person": "322-261-8260"
    },
    {
        "id": 27,
        "first_name": "Karyl",
        "last_name": "Weeds",
        "email": "kweedsq@moonfruit.com",
        "gender": "Female",
        "address": "PO Box 23578",
        "phone": "471-344-5419",
        "car": "XT",
        "close_person": "905-518-1949"
    },
    {
        "id": 28,
        "first_name": "Nahum",
        "last_name": "Rubinow",
        "email": "nrubinowr@uol.com.br",
        "gender": "Polygender",
        "address": "PO Box 17458",
        "phone": "221-925-8184",
        "car": "Hombre Space",
        "close_person": "224-178-1581"
    },
    {
        "id": 29,
        "first_name": "Maxine",
        "last_name": "Dast",
        "email": "mdasts@sitemeter.com",
        "gender": "Female",
        "address": "Apt 1302",
        "phone": "158-611-3095",
        "car": "350Z",
        "close_person": "939-348-3063"
    },
    {
        "id": 30,
        "first_name": "Carmelita",
        "last_name": "Pitman",
        "email": "cpitmant@about.com",
        "gender": "Female",
        "address": "Apt 1572",
        "phone": "726-301-4562",
        "car": "Cavalier",
        "close_person": "443-157-5706"
    },
    {
        "id": 31,
        "first_name": "Gaby",
        "last_name": "Prendergast",
        "email": "gprendergastu@squarespace.com",
        "gender": "Male",
        "address": "19th Floor",
        "phone": "201-286-5111",
        "car": "S80",
        "close_person": "404-500-9837"
    },
    {
        "id": 32,
        "first_name": "Loree",
        "last_name": "Blindermann",
        "email": "lblindermannv@angelfire.com",
        "gender": "Female",
        "address": "Room 1791",
        "phone": "619-402-2310",
        "car": "960",
        "close_person": "789-613-6710"
    },
    {
        "id": 33,
        "first_name": "Tiena",
        "last_name": "Daice",
        "email": "tdaicew@senate.gov",
        "gender": "Female",
        "address": "Apt 730",
        "phone": "537-202-8101",
        "car": "Mountaineer",
        "close_person": "846-539-9698"
    },
    {
        "id": 34,
        "first_name": "Barty",
        "last_name": "Queree",
        "email": "bquereex@delicious.com",
        "gender": "Male",
        "address": "19th Floor",
        "phone": "843-375-7958",
        "car": "Amigo",
        "close_person": "433-359-7977"
    },
    {
        "id": 35,
        "first_name": "Malva",
        "last_name": "Keuneke",
        "email": "mkeunekey@prlog.org",
        "gender": "Female",
        "address": "Room 1501",
        "phone": "526-267-2542",
        "car": "Explorer Sport",
        "close_person": "473-622-4656"
    },
    {
        "id": 36,
        "first_name": "Cyrillus",
        "last_name": "Keig",
        "email": "ckeigz@who.int",
        "gender": "Male",
        "address": "Apt 343",
        "phone": "207-950-2303",
        "car": "Virage",
        "close_person": "474-128-5896"
    },
    {
        "id": 37,
        "first_name": "Laurice",
        "last_name": "Roch",
        "email": "lroch10@webnode.com",
        "gender": "Female",
        "address": "Apt 1200",
        "phone": "218-741-9218",
        "car": "Defender 90",
        "close_person": "622-489-7691"
    },
    {
        "id": 38,
        "first_name": "Bernette",
        "last_name": "Atthow",
        "email": "batthow11@nih.gov",
        "gender": "Female",
        "address": "Apt 1760",
        "phone": "108-671-3618",
        "car": "Justy",
        "close_person": "327-410-1616"
    },
    {
        "id": 39,
        "first_name": "Yorgo",
        "last_name": "Whitehouse",
        "email": "ywhitehouse12@scribd.com",
        "gender": "Male",
        "address": "12th Floor",
        "phone": "400-910-4543",
        "car": "S4",
        "close_person": "502-490-5335"
    },
    {
        "id": 40,
        "first_name": "Merrily",
        "last_name": "O'Lunny",
        "email": "molunny13@npr.org",
        "gender": "Polygender",
        "address": "Suite 27",
        "phone": "257-347-0787",
        "car": "M5",
        "close_person": "484-201-7184"
    },
    {
        "id": 41,
        "first_name": "Kippie",
        "last_name": "de Cullip",
        "email": "kdecullip14@biblegateway.com",
        "gender": "Genderqueer",
        "address": "Apt 1506",
        "phone": "342-661-8501",
        "car": "Liberty",
        "close_person": "253-915-6146"
    },
    {
        "id": 42,
        "first_name": "Shawna",
        "last_name": "Ferrea",
        "email": "sferrea15@tinyurl.com",
        "gender": "Female",
        "address": "Room 519",
        "phone": "812-946-7680",
        "car": "Silverado",
        "close_person": "542-740-3127"
    },
    {
        "id": 43,
        "first_name": "Danit",
        "last_name": "Klimczak",
        "email": "dklimczak16@usatoday.com",
        "gender": "Female",
        "address": "PO Box 98825",
        "phone": "124-871-2062",
        "car": "Canyon",
        "close_person": "991-168-4391"
    },
    {
        "id": 44,
        "first_name": "Fernando",
        "last_name": "Francescuccio",
        "email": "ffrancescuccio17@mapy.cz",
        "gender": "Male",
        "address": "PO Box 63869",
        "phone": "737-368-0223",
        "car": "X-Type",
        "close_person": "430-838-9977"
    },
    {
        "id": 45,
        "first_name": "Dorthea",
        "last_name": "Dammarell",
        "email": "ddammarell18@smugmug.com",
        "gender": "Female",
        "address": "8th Floor",
        "phone": "988-420-8645",
        "car": "Maxima",
        "close_person": "527-250-4872"
    }
]