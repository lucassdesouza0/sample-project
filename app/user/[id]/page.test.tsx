import React from "react";
import { render } from "@testing-library/react";
import Page from "./page";
import { expect } from "@jest/globals";
import { Providers } from "@/lib/providers";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

const ReduxWrapper = ({ children }: { children?: React.ReactNode }) => {
  return <Providers>{children}</Providers>;
};

describe("Test", () => {
  it("should match the snapshot", () => {
    const tree = render(
      <ReduxWrapper>
        <Page params={{ id: 1 }} />
      </ReduxWrapper>
    );

    expect(tree).toMatchSnapshot();
  });

  it("should render the correct title", () => {
    const { getByText } = render(
      <ReduxWrapper>
        <Page params={{ id: 1 }} />
      </ReduxWrapper>
    );

    expect(getByText("User Details")).toBeTruthy();
  });
});
