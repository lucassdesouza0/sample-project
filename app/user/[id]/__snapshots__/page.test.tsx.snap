// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root MuiBox-root-1"
      >
        <h1>
          User Details
        </h1>
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg"
        >
          <table
            class="MuiTable-root"
          >
            <thead
              class="MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  First Name
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Last Name
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Email
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Phone Number
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Address
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Gender
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Car
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head"
                  scope="col"
                >
                  Close Person
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body"
                />
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root MuiBox-root-1"
    >
      <h1>
        User Details
      </h1>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg"
      >
        <table
          class="MuiTable-root"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                First Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Last Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Email
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Phone Number
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Address
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Gender
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Car
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head"
                scope="col"
              >
                Close Person
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
              <td
                class="MuiTableCell-root MuiTableCell-body"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
